translator/translate/otel/exporter/awsemf/translator.go

```diff
@@ -157,6 +171,10 @@ func isPrometheus(conf *confmap.Conf) bool {
 	return conf.IsSet(prometheusBasePathKey)
 }
 
+func setAppSignalsFields(_ *confmap.Conf, _ *awsemfexporter.Config) error {
```

Comment:
Why do we need this function?

Reply:
I originally had `setDisableMetricExtraction(ecsBasePathKey, conf, cfg)` and then removed it as I wasnt sure if it disabling metrics extraction from EMF logs should be an option for APM.

Thought Id just leave this for anything we might add in the future.

More importantly, in the place this is being called, I need to short circuit that one way or another. The ordering there kinda matters and it shouldnt hit isKubernetes when translating the emfexporter for appsignals - since its common for both app_signals and kubernetes to exist in the same config. And the k8s one will overwrite with unintended changes for the appsignals emfexporter section.

If someone does change that logic, our unit tests will complain.